packages/shared/client/ReplayClient.ts

```diff
@@ -877,75 +879,51 @@ export class ReplayClient implements ReplayClientInterface {
     }
   }
 
-  async _waitForPointToBeLoaded(point: ExecutionPoint): Promise<void> {
+  async waitForPointToBeInFocusRange(point: ExecutionPoint): Promise<void> {
     return new Promise(resolve => {
-      const checkLoaded = () => {
-        const loadedRegions = this.loadedRegions;
-        let isLoaded = false;
-        if (loadedRegions !== null) {
-          isLoaded = isPointInRegions(point, loadedRegions.loaded);
+      const checkFocusRange = () => {
+        let isInFocusRange = false;
+        if (this.focusRange !== null) {
+          isInFocusRange = isPointInRegion(point, this.focusRange);
         }
 
-        if (isLoaded) {
+        if (isInFocusRange) {
           resolve();
 
-          this.removeEventListener("loadedRegionsChange", checkLoaded);
+          this.removeEventListener("focusRangeChange", checkFocusRange);
         }
       };
 
-      this.addEventListener("loadedRegionsChange", checkLoaded);
+      this.addEventListener("focusRangeChange", checkFocusRange);
 
-      checkLoaded();
+      checkFocusRange();
     });
   }
 
-  async _waitForRangeToBeLoaded(
+  async waitForRangeToBeInFocusRange(
     focusRange: TimeStampedPointRange | PointRange | null
   ): Promise<void> {
-    return new Promise(resolve => {
-      const checkLoaded = () => {
-        const loadedRegions = this.loadedRegions;
-        let isLoaded = false;
-        if (loadedRegions !== null && loadedRegions.loading.length > 0) {
-          if (focusRange !== null) {
-            isLoaded = isRangeInRegions(focusRange, loadedRegions.indexed);
-          } else {
-            isLoaded = areRangesEqual(loadedRegions.indexed, loadedRegions.loading);
-          }
-        }
-
-        if (isLoaded) {
-          resolve();
-
-          this.removeEventListener("loadedRegionsChange", checkLoaded);
-        }
-      };
-
-      this.addEventListener("loadedRegionsChange", checkLoaded);
-
-      checkLoaded();
-    });
-  }
+    if (focusRange === null) {
+      return Promise.resolve();
```

Comment:
This path seems wrong, at a glance.

I think it's relying on the fact that if there's no explicit focus range, the whole recording is "focused" – but that's not true for longer recordings.

I think a more robust fix would be to remove this special case, and update the `getFocusWindow` method to also dispatch the new '"focusRangeChange"` event?

Reply:
> This path seems wrong, at a glance.

This references the `focusRange` parameter instead of the current focus range (i.e. `this.focusRange`), so it's correct. I've renamed the `focusRange` parameter to `range` now to avoid that confusion.

> update the getFocusWindow method to also dispatch the new '"focusRangeChange"` event

Good point! I've added that and also call `getFocusWindow()` in `configure()` and `initialize()` to ensure we always set the initial focus range.